packages/ec-core/src/operator/selector/recursive_weighted.rs

```diff
@@ -87,154 +130,95 @@ where
     }
 }
 
-pub struct WeightedSelectorPair<P, A, B>
-where
-    P: Population,
-    A: WeightedSelector<P>,
-    B: WeightedSelector<P>,
-{
+pub struct WeightedSelectorPair<A, B> {
     a: A,
     b: B,
-    // weight: usize,
-    _p: PhantomData<P>,
+    distr: Option<Bernoulli>,
+    weight_sum: u32,
```

Comment:
Remind me again why you prefer `u32` over `usize` here? I'm not sure I care, but I am curious.

Reply:
using `u32` over `usize` is probably more than we need ( we don't really need `u64` on 64-bit platforms and weights aren't really dependent on memory size or pointer width or anything so using usize in the first place doesn't really make sense - also using u32 allows us to use the rand bernouli distribution